File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/go_solutions/2432-the-employee-that-worked-on-the-longest-task.go
func hardestWorker(n int, logs [][]int) int {
    maxDuration := 0
    resultId := logs[0][0] // Initialize with the first employee's ID
    prevFinishTime := 0

    for _, log := range logs {
        id := log[0]
        finishTime := log[1]

        currentDuration := finishTime - prevFinishTime

        if currentDuration > maxDuration {
            maxDuration = currentDuration
            resultId = id
        } else if currentDuration == maxDuration {
            // If durations are equal, choose the employee with the smallest ID
            if id < resultId {
                resultId = id
            }
        }
        prevFinishTime = finishTime
    }

    return resultId
}

File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/c_solutions/2432-the-employee-that-worked-on-the-longest-task.c
int hardestWorker(int n, int** logs, int logsSize, int* logsColSize) {
    int maxDuration = logs[0][1];
    int resultEmployeeId = logs[0][0];
    int previousFinishTime = logs[0][1];

    for (int i = 1; i < logsSize; i++) {
        int currentEmployeeId = logs[i][0];
        int currentFinishTime = logs[i][1];
        int currentDuration = currentFinishTime - previousFinishTime;

        if (currentDuration > maxDuration) {
            maxDuration = currentDuration;
            resultEmployeeId = currentEmployeeId;
        } else if (currentDuration == maxDuration) {
            if (currentEmployeeId < resultEmployeeId) {
                resultEmployeeId = currentEmployeeId;
            }
        }
        previousFinishTime = currentFinishTime;
    }

    return resultEmployeeId;
}